docs/make.jl

```diff
@@ -14,4 +14,20 @@ makedocs(
     ],
 )
 
-deploydocs(repo="github.com/JuliaDiff/ChainRules.jl.git")
+const repo=="github.com/JuliaDiff/ChainRules.jl.git"
+if get(ENV, "TRAVIS_PULL_REQUEST", "false") == "false"
+    # Normal Case
+    deploydocs(repo=repo)
+else
+    @info "Deploying review docs for PR #$PR"
+    # Overwrite Documenter's function for generating the versions.js file
+    foreach(Base.delete_method, methods(Documenter.Writers.HTMLWriter.generate_version_file))
```

Comment:
🤢

Reply:
Can we link to an issue and make clear that this is evil

Reply:
also can we use `invoke` to get the same behaviour (slightly less evil)

Reply:
we can not use invoke because we are changing a mether that is called internally by Documentor.deploy docs.

I think real solution is for this to be added as a feature to Documentor,
and then we will just call that.

Reply:
Link to issue added

Reply:
This is basically just a "safer" version of overwriting/defining a more specific method.